lnwallet/rebroadcaster.go

```diff
@@ -0,0 +1,24 @@
+package lnwallet
+
+import (
+	"github.com/btcsuite/btcd/chaincfg/chainhash"
+	"github.com/btcsuite/btcd/wire"
+)
+
+// Rebroadcaster is an abstract rebroadcaster instance that'll continually
+// rebroadcast transactions in the background until they're confirmed.
+type Rebroadcaster interface {
+	// Starts launches all goroutines the rebroadcaster needs to operate.
+	Start() error
+
+	// Stop terminates the rebroadcaster and all goroutines it spawned.
+	Stop()
+
+	// Broadcast enqueues a transaction to be rebroadcast until it's been
+	// confirmed.
+	Broadcast(tx *wire.MsgTx) error
```

Comment:
Could enhance this to return a channel that delivers confirmation updates, perhaps it can help the caller write less code that's waiting on multiple channels.

Reply:
We have this already: https://github.com/lightningnetwork/lnd/blob/master/chainntnfs/interface.go#L229-L235. Not all sub-systems use it though.